src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java

```diff
@@ -144,7 +144,7 @@ public static void assignRequest(final IStandardRequestManager manager, final IR
                                         .filter(r -> r.getRequestType().isSupertypeOf(request.getRequestType()))
                                         .filter(r -> !failedResolvers.contains(r.getRequesterId()))
                                         .sorted(Comparator.comparing(r -> -1 * r.getPriority()))
-                                        .collect(Collectors.toSet());
+                                        .collect(Collectors.toCollection(LinkedHashSet::new));
 
             for (final IRequestResolver<?> resolver : resolversForRequestType)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS135)